KeyVault.Acmebot/Providers/GoDaddyProvider.cs

```diff
@@ -0,0 +1,207 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+using KeyVault.Acmebot.Options;
+
+using Newtonsoft.Json;
+
+namespace KeyVault.Acmebot.Providers
+{
+    public class GoDaddyProvider : IDnsProvider
+    {
+        public GoDaddyProvider(GoDaddyOptions options)
+        {
+            _client = new GoDaddyClient(options.ApiKey, options.ApiSecret, "https://api.godaddy.com");
```

Comment:
The endpoint URL should be embedded in the GoDaddyClient side, as it is unlikely to change.

```suggestion
            _client = new GoDaddyClient(options.ApiKey, options.ApiSecret);
```